File: Magic Triplet/code.java
class Solution{
    public int countTriplets(int[] nums){
        // code here
        int count=0;
        int n=nums.length;
        if(n<=2){
            return 0;
        }
        for(int i=1; i<n-1; i++){
            int sm=0; int lr=0;
            for(int j=0; j<i; j++){
                if(nums[j]<nums[i]){
                    sm++;
                }
            }
            for(int k=i+1; k<n; k++){
                if(nums[k]>nums[i]){
                    lr++;
                }
            }
            count+=sm*lr;
        }
        return count;
        
    }
}

File: Minimum X(xor)A/code.cpp
class Solution {

  public:

    int minVal(int a, int b) {

        // code here

        int ans=a;

        int ct_set=__builtin_popcount(b);

        int i=0;

        while(__builtin_popcount(ans)!=ct_set){

            // we have to find x such that they are closest so that their XOR can be minimum 

            // closer the number minimum the XOR

            // we have count of set bits in C

            // if count of set bits in a is greater than that in C then we have to reduce it

            //  this can be done by the below steps.

            if(__builtin_popcount(ans)>ct_set)ans&=(~(1<<i));

            // else increase the set bits count in ans

            else

            ans|=1<<i;

            i++;

        }

        // finally return ans

        

        return ans;

    }

};
